### ISSUE TITLE
Godot Editor should only change date modified on files when contents change

### ISSUE THREAD

**Issue by BenMcLean** [godotengine/godot#31186]
**Godot version:**
3.1.1.stable.mono.official

**OS/device including version:**
Windows 10 Home 64-Bit

**Issue description:**
***What happens***
I open my project and see that Git is reporting that project.godot, default_env.tres and Game.tscn (my root node) have changed despite their contents being identical.

***What should happen instead***
Last modified timestamps on files should only be changed when file contents change as is expected by Git and probably other source control systems.

**Steps to reproduce:**

Open any project.

**Minimal reproduction project:**
Any project.

But [here's the one I'm working on anyway](https://github.com/BenMcLean/Wolf3D-Godot)

**Comment by KoBeWi** [comment]
I think I encountered it too. Sometimes Git would flag files as changed, but the diff was empty. Doesn't seem to happen consistently tho.

**Comment by BenMcLean** [comment]
Oh ... maybe it only affects projects which programmatically create nodes at runtime?

**Comment by TicklesTheBrain** [comment]
I've not encountered this before, but whenever git says a file has changed but the diff is empty, the usual suspect is the line endings.

**Comment by akien-mga** [comment]
> I open my project and see that Git is reporting that project.godot, default_env.tres and Game.tscn (my root node) have changed despite their contents being identical.

Do you save the main scene after opening the project? If you save the scene, even if no changes were made, it's normal that the modified time changes for the scene, project.godot and related resources.

**Comment by BenMcLean** [comment]
No, I don't save the scene after opening the project. However, even if I did, last modified should not change if there were no changes to the contents of the file.

**Comment by OverseeingDev** [comment]
This is still happening as of 3.4.2.stable

**Comment by BenMcLean** [comment]
I think this may be a Windows issue.
